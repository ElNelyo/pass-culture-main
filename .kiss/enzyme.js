// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
