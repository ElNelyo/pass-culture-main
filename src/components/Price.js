import React from 'react'

const Price = ({ value }) => {
  return (
    <div className='price'>{
      value === 0
        ? 'gratuit'
        : value && (value.toString().replace('.', ',')+'â‚¬')
    }</div>
  )
}

export default Price;
