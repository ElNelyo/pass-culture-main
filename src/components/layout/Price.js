import React from 'react'
import PropTypes from 'prop-types'

import { getDisplayPrice } from '../../helpers'

const Price = ({ free, value, ...rest }) => {
  const price = getDisplayPrice(value, free)
  return (
    <span {...rest} className="price">
      {price}
    </span>
  )
}

Price.defaultProps = {
  free: null,
  value: null,
}

Price.propTypes = {
  free: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.array, PropTypes.number]),
}

export default Price
