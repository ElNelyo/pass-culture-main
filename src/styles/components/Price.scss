.price {
  min-height: 2rem;
}
