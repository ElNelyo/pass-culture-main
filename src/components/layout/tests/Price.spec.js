// jest ./src/components/layout/tests/Price --watch
import React from 'react'
import { shallow } from 'enzyme'

import Price from '../Price'

describe('src | components | pages | Price', () => {
  it('should match snapshot without props', () => {
    // given
    const props = {}
    // when
    const wrapper = shallow(<Price {...props} />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
  it('should match snapshot with a price number', () => {
    // given
    const props = { value: 5 }
    // when
    const wrapper = shallow(<Price {...props} />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
  it('should match snapshot with an array of prices number', () => {
    // given
    const props = { value: [5, 10] }
    // when
    const wrapper = shallow(<Price {...props} />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
  it('should match snapshot with a classname', () => {
    // given
    const props = { className: 'fake-css-class', value: [5, 10] }
    // when
    const wrapper = shallow(<Price {...props} />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
  it('should match snapshot free prop defined and a zero price', () => {
    // given
    const props = { free: 'Offre gratuite', value: 0 }
    // when
    const wrapper = shallow(<Price {...props} />)
    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})
