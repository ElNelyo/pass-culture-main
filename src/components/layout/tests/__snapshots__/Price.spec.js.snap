// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Price should match snapshot free prop defined and a zero price 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    className=""
    free="Offre gratuite"
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "0€",
      "className": "price ",
    },
    "ref": null,
    "rendered": "0€",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "0€",
        "className": "price ",
      },
      "ref": null,
      "rendered": "0€",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`src | components | pages | Price should match snapshot with a classname 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    className="fake-css-class"
    free="Gratuit"
    value={
      Array [
        5,
        10,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "5€ - 10€",
      "className": "price fake-css-class",
    },
    "ref": null,
    "rendered": "5€ - 10€",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "5€ - 10€",
        "className": "price fake-css-class",
      },
      "ref": null,
      "rendered": "5€ - 10€",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`src | components | pages | Price should match snapshot with a price number 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    className=""
    free="Gratuit"
    value={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "5€",
      "className": "price ",
    },
    "ref": null,
    "rendered": "5€",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "5€",
        "className": "price ",
      },
      "ref": null,
      "rendered": "5€",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`src | components | pages | Price should match snapshot with an array of prices number 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    className=""
    free="Gratuit"
    value={
      Array [
        5,
        10,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "5€ - 10€",
      "className": "price ",
    },
    "ref": null,
    "rendered": "5€ - 10€",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "5€ - 10€",
        "className": "price ",
      },
      "ref": null,
      "rendered": "5€ - 10€",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`src | components | pages | Price should match snapshot without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    className=""
    free="Gratuit"
    value={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "price ",
    },
    "ref": null,
    "rendered": "",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "price ",
      },
      "ref": null,
      "rendered": "",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
